crates/jxl-frame/src/lib.rs

```diff
@@ -31,6 +31,7 @@ enum GroupInstr {
 
 #[derive(Debug)]
 pub struct Frame<'a> {
+    idx: Option<usize>,
```

Comment:
IMO frame indices are better tracked in `jxl-render` as it's rendering specific state. Something like this...

```rs
struct IndexedFrame<'f> {
    frame: Frame<'f>,
    index: usize,
}
```

...and passing `&IndexedFrame<'_>` all around.